### ISSUE TITLE
Can't match at root with DSL

### ISSUE THREAD

**Issue by ajhcpt** [DiUS/pact-jvm#207]
Hi, I would like to be able to match at the body root on a response. In my case I'm trying to match a response body that doesn't have an attribute name just an integer:

I've setup the pact as follows: 

val pact = ConsumerPactBuilder.consumer("complex-instruction-service")
      .hasPactWith("workflow-service")
      .given("any state")
      .uponReceiving("a request to start a workflow")
        .path("/startWorkflowProcessInstance")
        .headers(headers)
        .method("POST")
        .body(requestBody)
      .willRespondWith()
        .headers(headers)
        .status(HttpStatusCodes.Ok)
        .body("1234")
      .toFragment

This results in the provider test failing unless there is an exact match on "1234". Can I somehow set this up as a regular expression? If the response had an attribute name I could use stringMatcher("name", regex).

Generated pact file:

{
    "providerState" : "any state",
    "description" : "a request to start a workflow",
    "request" : {
      "method" : "POST",
      "path" : "/startWorkflowProcessInstance",
      "headers" : {
        "Content-Type" : "application/json;odata=verbose"
      },
      "query" : "",
      "body" : {
        "workflowProcessName" : "crzFrpFJWRhqQdhtetrE",
        "variables" : [ {
          "name" : "VXHgTGCBOlFXWCUQXgQN",
          "value" : "ZQKXFxiIWBaWIpxEBZwl"
        } ]
      },
      "matchingRules" : {
        "$.body.variables" : {
          "min" : 1,
          "match" : "type"
        },
        "$.body.variables[*].value" : {
          "match" : "type"
        },
        "$.body.variables[*].name" : {
          "match" : "type"
        },
        "$.body.workflowProcessName" : {
          "match" : "type"
        }
      }
    },
    "response" : {
      "status" : 200,
      "headers" : {
        "Content-Type" : "application/json;odata=verbose"
      },
      "body" : 1234
    }
  } ],
  "metadata" : {
    "pact-specification" : {
      "version" : "2.0.0"
    },
    "pact-jvm" : {
      "version" : "2.3.2"
    }
  }
}

**Comment by uglyog** [comment]
I've added a new DSL class for matching values at the root. You will be able to the following:

```scala
val pact = ConsumerPactBuilder.consumer("complex-instruction-service")
    .hasPactWith("workflow-service")
    .given("any state")
    .uponReceiving("a request to start a workflow")
        .path("/startWorkflowProcessInstance")
        .headers(headers)
        .method("POST")
        .body(requestBody)
    .willRespondWith()
        .headers(headers)
        .status(HttpStatusCodes.Ok)
        .body(PactDslJsonRootValue.numberType("1234"))
    .toFragment
```

**Comment by uglyog** [comment]
This has been released with version 3.2.2
